From 8f7a015a83de3ec5df32ab881e49871bc1d3eaae Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Fri, 24 Nov 2023 08:57:36 +0100
Subject: [PATCH 0095/1695] dts: stm32mp15: disable non-secure IWDG2 on ST
 boards

Disable non-secure IWDG2 watchdog device in ST boards stm32mp157a-dk1,
stm32mp157c-dk2, stm32mp157c-ed1 and stm32mp157c-ev1. This watchdog is
fully under control of the non-secure world and OP-TEE is not expected
to interfere with it.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 671c1e6b3e25164fd5ef032df352e303801463f2 dts: stm32mp15: disable non-secure IWDG2 on ST boards]
Change-Id: I366abf67b686a0b182a649c4efc7af46dfc49b61
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
[etienne: cherry-picked from c825ffc94e2e878c70f0339a7bdf88bd39644f75]
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
---
 core/arch/arm/dts/stm32mp157c-ed1.dts  | 3 +--
 core/arch/arm/dts/stm32mp15xx-dkx.dtsi | 3 +--
 2 files changed, 2 insertions(+), 4 deletions(-)

diff --git a/core/arch/arm/dts/stm32mp157c-ed1.dts b/core/arch/arm/dts/stm32mp157c-ed1.dts
index 8663287ab..cee05c472 100644
--- a/core/arch/arm/dts/stm32mp157c-ed1.dts
+++ b/core/arch/arm/dts/stm32mp157c-ed1.dts
@@ -319,8 +319,7 @@
 
 &iwdg2 {
 	timeout-sec = <32>;
-	status = "okay";
-	secure-status = "disabled";
+	status = "disabled";
 };
 
 &m4_rproc {
diff --git a/core/arch/arm/dts/stm32mp15xx-dkx.dtsi b/core/arch/arm/dts/stm32mp15xx-dkx.dtsi
index 87f853b3f..7ae62e7c0 100644
--- a/core/arch/arm/dts/stm32mp15xx-dkx.dtsi
+++ b/core/arch/arm/dts/stm32mp15xx-dkx.dtsi
@@ -457,8 +457,7 @@
 
 &iwdg2 {
 	timeout-sec = <32>;
-	status = "okay";
-	secure-status = "disabled";
+	status = "disabled";
 };
 
 &ltdc {
-- 
2.47.2

